### ISSUE TITLE
TSC runs out of memory when TransportStream videos are present in the path

### ISSUE THREAD

**Issue by Ved-s** [microsoft/TypeScript#56516]
### 🔎 Search Terms

typescript transportstream

### 🕗 Version & Regression Information

- This is bug that causes OOM
- This is the behavior in every version I tried
- This behavior exists on previous versions, checked on 4.2.4

### 💻 Code

```ts
// Your code here
```

### 🙁 Actual behavior

I had a 300MB TransportStream (`video.ts`) video in my project dir, this caused tsc to try and process it, which caused an out-of-memory exception.

### 🙂 Expected behavior

tsc should ignore non-text files with `.ts` extensions, or blacklist videos or TransportStream specifically.

**Comment by jakebailey** [comment]
Did you only check 4.2.4, or have you checked the latest version? More recent versions should be ignoring binary files.

**Comment by Ved-s** [comment]
@jakebailey I checked on latest that installs with `npm -i typescript` on node 20

**Comment by nmain** [comment]
> More recent versions (as of https://github.com/microsoft/TypeScript/pull/53667) should be ignoring binary files.

Only checking the first character isn't going to be sufficient for arbitrary binary files.  A perfectly random sequence of bytes only has a ~47% chance to fail UTF8 decoding on the first character, and a valid MPEG-TS file has a 0% chance to fail UTF8 decoding (the first byte is always ASCII "G").

**Comment by jakebailey** [comment]
@Ved-s @nmain Can you try the build on https://github.com/microsoft/TypeScript/pull/57008#issuecomment-1885734113 and see if that fixes your issue?

**Comment by Ved-s** [comment]
```
video.ts:1:1 - error TS1490: File appears to be binary.


Found 1 error in video.ts:1


real	0m7,533s
user	0m7,626s
sys	0m0,379s
``` it detects is as binary now, but it takes time to detect it, 8 seconds on just `tsc` (compile all) and 11s on `tsc --watch` on the first compile, then it compiles instantly, ignoring the video file

**Comment by jakebailey** [comment]
Yeah, this pass happens after the entire file is read. Well, at least it's not an OOM, so that's better.

**Comment by jakebailey** [comment]
Of course, you should just exclude that file from the project 😄
